tests/system/libraries/SystemTestSpy/speechSpyGlobalPlugin.py

```diff
@@ -316,33 +319,82 @@ def get_next_speech_index(self) -> int:
 		"""
 		return self.get_last_speech_index() + 1
 
-	def wait_for_specific_speech(
+	def _has_speech_occurred_before_timeout(
 			self,
 			speech: str,
-			afterIndex: Optional[int] = None,
-			maxWaitSeconds: int = 5,
-	) -> int:
+			afterIndex: Optional[int],
+			maxWaitSeconds: float,
+			intervalBetweenSeconds: float,
+	) -> Tuple[bool, Optional[int]]:
 		"""
 		@param speech: The speech to expect.
 		@param afterIndex: The speech should come after this index. The index is exclusive.
 		@param maxWaitSeconds: The amount of time to wait in seconds.
-		@return: the index of the speech.
+		@param intervalBetweenSeconds: The amount of time to wait between checking speech, in seconds.
+		@return: True if the speech occurred and the index of the speech.
 		"""
-		success, speechIndex = _blockUntilConditionMet(
+		return _blockUntilConditionMet(
 			getValue=lambda: self._getIndexOfSpeech(speech, afterIndex),
 			giveUpAfterSeconds=self._minTimeout(maxWaitSeconds),
 			shouldStopEvaluator=lambda indexFound: indexFound >= (afterIndex if afterIndex else 0),
-			intervalBetweenSeconds=0.1,
+			intervalBetweenSeconds=intervalBetweenSeconds,
 			errorMessage=None
 		)
+
+	def wait_for_specific_speech(
+			self,
+			speech: str,
+			afterIndex: Optional[int] = None,
+			maxWaitSeconds: float = 5.0,
+			intervalBetweenSeconds: float = 0.1,
+	) -> int:
+		"""
+		@param speech: The speech to expect.
+		@param afterIndex: The speech should come after this index. The index is exclusive.
+		@param maxWaitSeconds: The amount of time to wait in seconds.
+		@param intervalBetweenSeconds: The amount of time to wait between checking speech, in seconds.
+		@return: the index of the speech.
+		"""
+		success, speechIndex = self._has_speech_occurred_before_timeout(
+			speech,
+			afterIndex,
+			maxWaitSeconds,
+			intervalBetweenSeconds
+		)
 		if not success:
 			self.dump_speech_to_log()
 			raise AssertionError(
-				"Specific speech did not occur before timeout: {}\n"
-				"See NVDA log for dump of all speech.".format(speech)
+				f"Specific speech did not occur before timeout: {speech}\n"
+				"See NVDA log for dump of all speech."
 			)
 		return speechIndex
 
+	def ensure_speech_did_not_occur(
+			self,
+			speech: str,
+			afterIndex: Optional[int] = None,
+			maxWaitSeconds: float = 5.0,
+			intervalBetweenSeconds: float = 0.1,
+	) -> None:
+		"""
+		@param speech: The speech to check for.
+		@param afterIndex: Check for speech after this index. The index is exclusive.
+		@param maxWaitSeconds: The amount of time to wait in seconds.
+		@param intervalBetweenSeconds: The amount of time to wait between checking speech, in seconds.

```

Comment:
It's probably worth putting a warning on this; the 'expected path' requires delaying the test by `maxWaitSeconds`. Generally, the system tests should be constructed so that they minimize any required waiting periods. Most of the time speech should start and be completed within tenths of a second.

Reply:
I considered this problem initially and wasn't sure of a good solution, or if it was a major problem.
I'm going to also make the parameter required instead of optional, so you have to explicitly consider and set the expected delay of the test.

Reply:
Yep, I think there are cases where it would be hard to avoid a check like this. As long as the developer is aware of it, and understand the trade off they are making.

It might also be worth pointing to the default for "has speech completed", which has to wait some amount of time, mostly to be sure that the application NVDA is working with / the OS doesn't have events pending.